### ISSUE TITLE
The new app / lib templates should help prevent checking in msbuild binlog files

### ISSUE THREAD

**Issue by jonthysell** [microsoft/react-native-windows#14160]
### Summary

1. The `init-windows` for the new templates should inject `*.binlog` into the project root `.gitignore` if the ignore file exists (and the line doesn't exist)
2. (Optional): We shouldn't dump the binlog to the root folder, maybe we could put it somewhere else (maybe the windows folder, where it's easier for us to gitignore it) and then output the path to the log file at the end of the command.

### Motivation

Right now, if you run `run-windows --logging` you'll get an `msbuild.binlog` file in the root of your project file (where you run `run-windows` 99% of the time). This file risks being checked in accidentally.

**Comment by chrisglein** [comment]
Related: https://github.com/microsoft/react-native-windows/issues/5937
Issue there was the binlog getting dropped in the root and having different rules for what the gitignore could do. Moving where it got dropped would help.

**Comment by jonthysell** [comment]
Changes for this will most likely be in `vnext\templates\templateUtils.js` to be used by the templateConfig.js files in the cpp-lib and cpp-app templates.

@jonthysell for more help/guidance.

**Comment by frankcalise** [comment]
@jonthysell Looking at helping with this issue.

1. Seems straight forward enough, would you do something like that inside the cli at `initWindows.ts`?
2. Going this route would avoid having to do 1 at all, do we just want to default `buildLogDirectory` to the `windows` dir, then update the vnext ignore templates?

I suppose we would want to add `*.err` and `*.wrn` in addition to `*.binlog`. Thoughts?

![Image](https://github.com/user-attachments/assets/07f0b602-c99e-4e18-bafc-147e11d2d104)

**Comment by jonthysell** [comment]
> [@jonthysell](https://github.com/jonthysell) Looking at helping with this issue.
> 
> 1. Seems straight forward enough, would you do something like that inside the cli at `initWindows.ts`?
> 2. Going this route would avoid having to do 1 at all, do we just want to default `buildLogDirectory` to the `windows` dir, then update the vnext ignore templates?
> 
> I suppose we would want to add `*.err` and `*.wrn` in addition to `*.binlog`. Thoughts?
> 
> ![Image](https://github.com/user-attachments/assets/07f0b602-c99e-4e18-bafc-147e11d2d104)

I agree if we just did 2 that would give us what we need and not pollute the project's main folder with build output. It's interesting to me that if you **don't** specify a folder with `--buildLogDirectory`, we only keep the binlog file and then put the err and wrn files in the temp folder.

I think then, the fix looks something like this in `msbuildttools.ts`:

1. Update the `logPrefix` to default to the `windows` folder (rather than the tmp folder) if `buildLogDirectory` is not specified: https://github.com/microsoft/react-native-windows/blob/a4e227eb668feae3a9696f339432dfbb1d1f9878/packages/%40react-native-windows/cli/src/utils/msbuildtools.ts#L96-L99
2. Remove the `localBinLog` (I don't see why we need that at all) and update the `const binlog` to
```ts
const binlog = `:${logPrefix}.binlog`;
```
here: https://github.com/microsoft/react-native-windows/blob/a4e227eb668feae3a9696f339432dfbb1d1f9878/packages/%40react-native-windows/cli/src/utils/msbuildtools.ts#L104-L105

Then we just need to add `*.binlog`, `*.err`, and `*.wrn` to the template `_gitignore` files at [vnext/templates/cpp-app/windows/_gitignore](https://github.com/microsoft/react-native-windows/blob/main/vnext/templates/cpp-app/windows/_gitignore) and [vnext/templates/cpp-lib/windows/_gitignore](https://github.com/microsoft/react-native-windows/blob/main/vnext/templates/cpp-lib/windows/_gitignore)

**Comment by frankcalise** [comment]
@jonthysell thanks for the guidance! I have a draft PR up, I still want to test it locally better.

I was reading the [Contributing to the CLI](https://github.com/microsoft/react-native-windows/wiki/Contributing-to-the-CLI#where-stuff-lives) docs, but I didn't quite grok the proper way to test my local CLI changes just yet.
